vm/src/builtins/set.rs

```diff
@@ -605,7 +605,13 @@ impl PySet {
 
     #[pymethod(magic)]
     fn rsub(&self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult<PyArithmeticValue<Self>> {
-        self.sub(other, vm)
+        PySet::number_downcast(other.to_number()).sub(
+            PySet {
+                inner: self.inner.to_owned(),
+            }
+            .into_pyobject(vm),
+            vm,
+        )
```

Comment:
sub and rsub shares many things. the implementation will be similar.

```suggestion
        if let Ok(other) = AnySet::try_from_object(vm, other) {
            Ok(PyArithmeticValue::Implemented(other.op(
                self,
                PySetInner::difference,
                vm,
            )?))
        } else {
            Ok(PyArithmeticValue::NotImplemented)
        }
```

Reply:
Set implementation looks not ideal. and there was working unfinished PRs also.

Reply:
Otherwise, would this work?
```suggestion
        if let Ok(other) = AnySet::try_from_object(vm, other) {
            Ok(PyArithmeticValue::Implemented(other.inner.difference(self, vm))?))
        } else {
            Ok(PyArithmeticValue::NotImplemented)
        }
```

Reply:
This version will work
```rust
    #[pymethod(magic)]
    fn rsub(
        zelf: PyRef<Self>,
        other: PyObjectRef,
        vm: &VirtualMachine,
    ) -> PyResult<PyArithmeticValue<Self>> {
        Ok(if let Ok(other) = AnySet::try_from_object(vm, other) {
            PyArithmeticValue::Implemented(Self {
                inner: other
                    .as_inner()
                    .difference(ArgIterable::try_from_object(vm, zelf.into())?, vm)?,
            })
        } else {
            PyArithmeticValue::NotImplemented
        })
    }
```

Reply:
The above style has been adopted in the PR.